sources/Adapters/picoTracker/system/input.cpp

```diff
@@ -1,3 +1,19 @@
 #include "input.h"
+#include <stdio.h>
 
-uint16_t scanKeys() { return (~gpio_get_all() & 0x0001FF00) >> 8; }
+uint16_t scanKeys() {
+#ifdef USB_REMOTE_UI
+  // This reads a byte from USB serial input in non-blocking way, by using
+  // param of 0 for immediate timeout and check for return of 255 for no
+  // char read result ref: https://forums.raspberrypi.com/viewtopic.php?t=303964
+  // the byte is expected to be sent by the remote UI client in the same bitmask
+  // format as is returned by the picotracker hardware reading the gpio pins
+  // group by the code below.
+  // That bit mask is documented in the KEYPAD_BITS enum.
+  char c = getchar_timeout_us(0);
+  if (c != 255) {
+    return c;
```

Comment:
Is there no use case for accepting inputs from remote and local keys? Wouldn't it make sense to "and" both inputs?

Reply:
Hmm, I can't think of a reason for wanting to do both at once, do you mean like having one hand do arrow keys on a physical device and then pressing play button computer or something like that?

Reply:
@maks BTW, on that note. Is display working remote only on this mode or both?

The way I see it is that if eventually this is something that wants to be supported on the main version as an option, we may want to be consistent and the remote option be auxiliary to the main device functionality. But we can certainly do some intermediate step here.

Reply:
Ah I see! Actually that's not how this works. 
Because I use an immediate timeout value, if there is no char waiting on serial to be read, its a no op and then the GPIO value is read and returned. So you can actually use both at the same time, in testing I even held down the nav button while sending key buttons, but that was when the code used individual chars per button before I switched to needing the client to send whole bitmask.
I'll add a comment to explain this.